arlas-core/src/main/java/io/arlas/server/utils/ParamsParser.java

```diff
@@ -238,9 +240,24 @@ public static Filter getFilter(List<String> filters, List<String> q, List<String
         filter.notpwithin = getStringMultiFilter(notpwithin);
         filter.notgwithin = getStringMultiFilter(notgwithin);
         filter.notgintersect = getStringMultiFilter(notgintersect);
+        filter.dateformat = dateFormat;
         return filter;
     }
 
+    public static String parseDate(String dateValue, String dateFormat) throws ArlasException {
+        String parsedDate = dateValue;
+        if (!StringUtil.isNullOrEmpty(dateFormat)) {
+            try {
+                parsedDate = String.valueOf(DateTimeFormat.forPattern(dateFormat).withZoneUTC().parseDateTime(dateValue).getMillis());
+            } catch (DateTimeParseException e) {
+                throw new InvalidParameterException(dateValue + " doesn't match the date format '" + dateFormat + "'");
+            } catch (IllegalArgumentException e) {
+                throw new InvalidParameterException(dateValue + " doesn't match the date format '" + dateFormat + "'");
```

Comment:
Add getMessage from the exception so that we know what is wrong with that date & format.

Reply:
Done :+1: